frosthaven_assistant/lib/Layout/menus/add_character_menu.dart

```diff
@@ -240,4 +182,62 @@ class AddCharacterMenuState extends State<AddCharacterMenu> {
                       }))
             ])));
   }
+
+  Widget? characterTile(BuildContext context, int index) {
+    CharacterClass character = _foundCharacters[index];
+    bool characterUnlocked =
+        _gameState.unlockedClasses.contains(character.name);
+    bool characterAlreadyAdded = _characterAlreadyAdded(character.name);
+    bool characterIsObjective = character.name == "Objective";
+    bool characterIsEscort = character.name == "Escort";
+
+    return ListTile(
+      leading: Image(
+        height: 40,
+        width: 40,
+        fit: BoxFit.contain,
+        color: character.hidden && !characterUnlocked ||
+                characterIsEscort ||
+                characterIsObjective
```

Comment:
If we wanted to we could continue to pull things out of here to make it easier to read. I wouldn't be opposed to turning this into something like

```
bool shouldUseCharacterColor = character.hidden && !characterUnlocked || characterIsEscort || characterIsObjective
```

Then we could use that here. However I didn't want to go overboard with pulling out variables if this isn't the style you prefer.

Reply:
That is absolutely fine.

Further, It is I believe recommended by the flutter team to create new Widgets rather than returning Widgets from methods. I know I've not been on top of this, so there's a bit of both in the codebase. In any case it's more of a style thing, and to promote reusability of custom widgets.

Reply:
Ah, that makes sense. I'll take a stab at turning this into a proper widget then.

Reply:
Okay, I converted that into a proper widget. Would you prefer I pull that new widget into its own file as well?

Reply:
No need, since this is only used here. If you feel like it would make the code cleaner/clearer, you are welcome to do it anyway.

Reply:
Okay. Actually, now that you mention it that, I feel like if I were to make it stateless, change onCharacterAdded to a more generic onSelect, and add a disabled attribute, then we may be able to reuse it in when removing characters. Thoughts?

Reply:
Knock yourself out ;) 

I have put little to no effort into making the menus and menu widgets reusable, so there's clearly room for improvement.
there's also obvious similarities with add/remove monsters too for example, though the data is different.

Reply:
Alright @Tarmslitaren, I think this is ready for a final review. Let me know what you think.

Reply:
Looks good to me. I'll merge